src/coreclr/src/pal/src/libunwind/CMakeLists.txt

```diff
@@ -1,21 +1,125 @@
-add_subdirectory(src)
-
 # define variables for the configure_file below
 
-if (CLR_CMAKE_HOST_ARCH_AMD64)
-    set(arch x86_64)
-elseif(CLR_CMAKE_HOST_ARCH_ARM64)
-    set(arch aarch64)
-elseif(CLR_CMAKE_PLATFROM_ARCH_ARM)
-    set(arch arm)
-elseif(CLR_CMAKE_HOST_ARCH_I386)
-    set(arch x86)
-endif()
-
 set(PKG_MAJOR "1")
 set(PKG_MINOR "5")
 set(PKG_EXTRA "-rc2")
+if(CLR_CMAKE_HOST_UNIX)
+    if (CLR_CMAKE_HOST_ARCH_AMD64)
+      set(arch x86_64)
+    elseif(CLR_CMAKE_HOST_ARCH_ARM64)
+      set(arch aarch64)
+    elseif(CLR_CMAKE_PLATFROM_ARCH_ARM)
+      set(arch arm)
+    elseif(CLR_CMAKE_HOST_ARCH_I386)
+        set(arch x86)
+    endif ()
+endif(CLR_CMAKE_HOST_UNIX)
+
+if(CLR_CMAKE_HOST_WIN32)
+    if (CLR_CMAKE_TARGET_ARCH_AMD64)
+      set(TARGET_AMD64 1)
+      set(arch x86_64)
+      add_definitions(-D__x86_64__)
+      add_definitions(-D__amd64__)
+    elseif(CLR_CMAKE_TARGET_ARCH_ARM64)
+      set(TARGET_AARCH64 1)
+      set(arch aarch64)
+      add_definitions(-D__aarch64__)
+    elseif(CLR_CMAKE_TARGET_ARCH_ARM)
+      set(TARGET_ARM 1)
+      set(arch arm)
+      add_definitions(-D__arm__)
+    else ()
+      message(FATAL_ERROR "Unrecognize value in environment variable TARGET")
+    endif ()
+
+    set(PACKAGE_STRING "libunwind")
```

Comment:
I was struggling to deal with the new problem that upstream now has CMakeList files.  I wanted to move all of our custom make stuff to a single file.  So my initial draft moved most of the Unix build stuff to the top level libunwind directory.  However it was never pure.

When I had trouble building on arm64, I started reverting Unix changes.  In the end, the Unix changes ended up being mostly indentation.

I put the Windows stuff here, because that is most similar to upstream.

It feels like compiler configuration belongs at the top of the library, so I can move Unix here if you are OK with that.  

I am not sure if I can move the all Windows stuff into src.  Some of it is required for configure `libunwind.in`.